src/structured_config.py

```diff
@@ -0,0 +1,200 @@
+            )
+        return value
+
+    @validator("log_cleanup_policy")
+    @classmethod
+    def log_cleanup_policy_validator(cls, value: str) -> Optional[str]:
+        """Check validity of `log_cleanup_policy` field."""
+        try:
+            _log_cleanup_policy = LogCleanupPolicy(value)
+        except Exception as e:
+            raise ValueError(
+                f"Value out of the accepted values. Could not properly parsed the roles configuration: {e}"
+            )
+        return value
+
+    @validator("log_cleaner_min_compaction_lag_ms")
+    @classmethod
+    def log_cleaner_min_compaction_lag_ms_validator(cls, value: str) -> Optional[int]:
+        """Check validity of `log_cleaner_min_compaction_lag_ms` field."""
+        int_value = int(value)
+        if int_value >= 0 and int_value <= 1000 * 60 * 60 * 24 * 7:
+            return int_value
+        raise ValueError("Value out of range.")
+
+    @validator("log_cleaner_delete_retention_ms")
+    @classmethod
+    def log_cleaner_delete_retention_ms_validator(cls, value: str) -> Optional[int]:
+        """Check validity of `log_cleaner_delete_retention_ms` field."""
+        int_value = int(value)
+        if int_value > 0 and int_value <= 1000 * 60 * 60 * 24 * 90:
+            return int_value
+        raise ValueError("Value out of range.")
+
+    @validator("transaction_state_log_num_partitions", "offsets_topic_num_partitions")
+    @classmethod
+    def between_zero_and_10k(cls, value: int) -> Optional[int]:
+        """Check that the integer value is between zero and 10000."""
+        if value >= 0 and value <= 10000:
+            return value
+        raise ValueError("Value below zero or greater than 10000.")
+
+    @validator("log_retention_bytes", "log_retention_ms")
+    @classmethod
+    def greater_than_minus_one(cls, value: str) -> Optional[int]:
+        """Check value greater than -1."""
+        int_value = int(value)
+        if int_value < -1:
+            raise ValueError("Value below -1. Accepted values are greater or equal than -1.")
+        return int_value
+
+    @validator("log_flush_interval_messages", "log_flush_interval_ms")
+    @classmethod
+    def greater_than_one(cls, value: str) -> Optional[int]:
+        """Check value greater than one."""
+        int_value = int(value)
+        if int_value < 1:
+            raise ValueError("Value below 1. Accepted values are greater or equal than 1.")
+        return int_value
+
+    @validator("replication_quota_window_num", "log_segment_bytes", "message_max_bytes")
+    @classmethod
+    def greater_than_zero(cls, value: int) -> Optional[int]:
+        """Check value greater than zero."""
+        if value < 0:
+            raise ValueError("Value below 0. Accepted values are greater or equal than 0.")
+        return value
+
+    @validator("compression_type")
+    @classmethod
+    def value_compression_type(cls, value: str) -> Optional[str]:
```

Comment:
I thought that this was not needed. Was I wrong?

Reply:
Removed.